legate/util/settings.py

```diff
@@ -0,0 +1,307 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+""" Control global configuration options with environment variables.
+
+Precedence
+~~~~~~~~~~
+
+Setting values are always looked up in the following prescribed order:
+
+immediately supplied values
+    These are values that are passed to the setting:
+
+    .. code-block:: python
+
+        settings.consensus(value)
+
+    If ``value`` is not None, then it will be returned, as-is. Otherwise, if
+    None is passed, then the setting will continue to look down the search
+    order for a value. This is useful for passing optional function paramters
+    that are None by default. If the parameter is passed to the function, then
+    it will take precedence.
+
+previously user-set values
+    If the value is set explicity in code:
+
+    .. code-block:: python
+
+        settings.minified = False
+
+    Then this value will take precedence over other sources. Applications
+    may use this ability to set values supplied on the command line, so that
+    they take precedence over environment variables.
+
+environment variable
+    Values found in the associated environment variables:
+
+    .. code-block:: sh
+
+        LEGATE_CONSENSUS=yes legate script.py
+
+local defaults
+    These are default values defined when accessing the setting:
+
+    .. code-block:: python
+
+        settings.concensus(default=True)
+
+    Local defaults have lower precendence than every other setting mechanism
+    except global defaults.
+
+global defaults
+    These are default values defined by the setting declarations. They have
+    lower precedence than every other setting mechanism.
+
+If no value is obtained after searching all of these locations, then a
+RuntimeError will be raised.
+
+"""
+from __future__ import annotations
+
+import os
+from typing import Any, Generic, Type, TypeVar, Union
+
+from typing_extensions import TypeAlias
+
+__all__ = (
+    "convert_str",
+    "convert_bool",
+    "convert_str_seq",
+    "PrioritizedSetting",
+    "Settings",
+)
+
+
+class _Unset:
+    pass
+
+
+T = TypeVar("T")
+
+
+Unset: TypeAlias = Union[T, Type[_Unset]]
+
+
+def convert_str(value: str) -> str:
+    """Return a string as-is."""
+    return value
+
+
+def convert_bool(value: bool | str) -> bool:
+    """Convert a string to True or False.
+
+    If a boolean is passed in, it is returned as-is. Otherwise the function
+    maps the following strings, ignoring case:
+
+    * "yes", "1", "on" -> True
+    * "no", "0", "off" -> False
+
+    Args:
+        value (str):
+            A string value to convert to bool
+
+    Returns:
+        bool
+
+    Raises:
+        ValueError
+
+    """
+    if isinstance(value, bool):
+        return value
+
+    val = value.lower()
+    if val in ["yes", "1", "on", "true", "True"]:
```

Comment:
No need to check against "True", since you called `lower()`

Reply:
c5c0796